pkg/storage/mountpoint.go

```diff
@@ -46,7 +47,9 @@ func (s *MountPointStorage) Mount(localPath string) error {
 
 	output, err := s.mountCmd.CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("error: %+v, %s", err, string(output))
+		// Cleanup the temporary mountpoint directory if the mount fails
+		errors.Join(err, os.RemoveAll(localPath))
+		return fmt.Errorf("%+v, %s", err, string(output))
```

Comment:
What happens if os.RemoveAll doesn't return an error? 

I kind of think these should be separated. I think its more clear this way.

```go
removeErr := os.RemoveAll()
if removeErr != nil {
  err = errors.Join(err, removeError)
}

return err
```

Reply:
I changed to just ignore the `os.RemoveAll` per @luke-lombardi's recommendation